### DISCUSSION TITLE
Cannot connect to YugaByte

### DISCUSSION THREAD

**Discussion by mithali-zs** [community/community#26862] [Actions]
<p>Hi,</p>
<p>I have a Go application that uses YugaByte and I have some tests for the same. While I run the tests in the pipeline and try to connect to yugabyte, it says connection refused.</p>
<p>Here is a snippet from my workflow file:</p>
<pre><code class="lang-auto">services:
      yugabyte:
        image: yugabytedb/yugabyte:2.2.2.0-b15
        ports:
          - 2021:7000
          - 2022:9000
          - 2023:5433
          - 2024:9042
</code></pre>
<p>Locally, I use the following command to run the container:</p>
<pre><code class="lang-auto">docker run  -d -p2021:7000 -p2022:9000 -p2023:5433 -p2024:9042 -v ~/yb_data:/home/yugabyte/var yugabytedb/yugabyte:latest bin/yugabyted start --daemon=false
</code></pre>
<p>I am not sure how I can specify these docker commands in the workflow file.</p>
<p>Thank You</p>

**Comment by weide-zhou** [comment]
<p>Hi <a class="mention" href="/mithali-zs">@mithali-zs</a>,</p>
<p>Glad to see you in Github Commnity Forum!</p>
<p>The code works fine on my <a href="https://github.com/weide-zhou/ticket13/runs/1110021302?check_suite_focus=true#step:2:54" rel="nofollow noopener">workflow</a>. Could it be possible that the connection refused caused by the network issue? Please try again with Github hosted runner.</p>
<p>If the issue persists, could you please share a sample repo/workflow yaml for further investigation?</p>
<p>Please check github docker services doc and sample workflows below, hope it helps.<br>
<a href="https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#example-using-localhost" class="onebox" target="_blank" rel="nofollow noopener">https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#example-using-localhost</a><br>
</p><aside class="onebox allowlistedgeneric">
  <header class="source">
      <img src="https://github.githubassets.com/favicons/favicon.svg" class="site-icon" width="32" height="32">
      <a href="https://github.com/actions/example-services/tree/main/.github/workflows" target="_blank" rel="nofollow noopener">GitHub</a>
  </header>
  <article class="onebox-body">
    <img src="https://user-images.githubusercontent.com/52772761/181100343-f5c07fd0-54a4-4084-9e8f-d89e0ad5d35b.png" class="thumbnail onebox-avatar" width="400" height="400">

<h3><a href="https://github.com/actions/example-services/tree/main/.github/workflows" target="_blank" rel="nofollow noopener">actions/example-services</a></h3>

<p>Example workflows using service containers. Contribute to actions/example-services development by creating an account on GitHub.</p>


  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>

**Comment by mithali-zs** [comment]
<p>Hi <a class="mention" href="/weide-zhou">@weide-zhou</a>.</p>
<p>Thanks for your reply.</p>
<p>When I tried it out, the service container was up and running. But I am not able to connect to it in my code.</p>
<p>What I have noticed even on my local is that in order for me to create a cluster successfully, <code>bin/yugabyted start --daemon=false</code> needs to be specified in docker run, else the container stops and you don’t see it when you do <code>docker ps</code>.</p>
<p>Here is the link to the documentation I referred to : <a href="https://docs.yugabyte.com/latest/quick-start/create-local-cluster/docker/#1-create-a-local-cluster" rel="noopener nofollow ugc">https://docs.yugabyte.com/latest/quick-start/create-local-cluster/docker/#1-create-a-local-cluster</a></p>
<p>Is there a way for me to provide this in workflow file?</p>

**Comment by weide-zhou** [comment, marked as answer]
<p>Hi <a class="mention" href="/mithali-zs">@mithali-zs</a>,</p>
<p>This should be due to the service is not alive during workflow, hence you cannot connect it.<br>
Typically we can add a simple healthcheck to keep the service alive, and then you can connect it and run your docker command in the workflow step.<br>
Please refer to the answer <a href="https://github.community/t/cannot-connect-to-cassandra-db-from-my-application/123282/4">here</a> for more details.</p>
<p>BTW, there is an action ‘<a href="https://github.com/yugabyte/yugabyte-db-action" rel="noopener nofollow ugc">yugabyte/yugabyte-db-action@master</a>’ which can help to set up a YugabyteDB database, does it meet your requirements?</p>
<pre><code class="lang-auto">- name: Setup YugabyteDB cluster
  uses: yugabyte/yugabyte-db-action@master
  with:
    yb_image_tag: 2.2.0.0-b80
    yb_master_ui_port: 7000
    yb_tserver_ui_port: 9000    
    ysql_port: 5433
    ycql_port: 9042
</code></pre>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/5/5f4325a2a4795ba8df65ff3062e52260a14c27a1.png" data-download-href="/uploads/short-url/dAJdJsICL7sQgMLWqtRxZ5IjTTH.png?dl=1" title="image" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/52772761/181100347-dd861c55-46dd-4f02-afce-c6ab6b806b47.png" alt="image" data-base62-sha1="dAJdJsICL7sQgMLWqtRxZ5IjTTH" width="690" height="137" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/5/5f4325a2a4795ba8df65ff3062e52260a14c27a1_2_690x137.png, https://aws1.discourse-cdn.com/github/optimized/2X/5/5f4325a2a4795ba8df65ff3062e52260a14c27a1_2_1035x205.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/2X/5/5f4325a2a4795ba8df65ff3062e52260a14c27a1_2_1380x274.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/5/5f4325a2a4795ba8df65ff3062e52260a14c27a1_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">image</span><span class="informations">1560×310 20.5 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>Thanks</p>

**Comment by mithali-zs** [comment]
<p>Hi <a class="mention" href="/weide-zhou">@weide-zhou</a>,</p>
<p>Thank you for your help. :slight_smile:<br>
As you suggested, I tried to add a healthcheck to keep the service alive. That did not work. It said yugabyte was unhealthy.</p>
<p>However,   ‘<a href="https://github.com/yugabyte/yugabyte-db-action" rel="noopener nofollow ugc">yugabyte/yugabyte-db-action@master </a>’ works fine. But there is an issue here. This creates a docker container and it does not stop even after the pipeline ends. So when I re run the pipeline, it fails at Setup YugabyteDB cluster as it complains that the port is already being used.</p>
<p>Is there a cleaner way to do this?</p>
<p>Thank You</p>

**Comment by weide-zhou** [comment]
<p>Hi <a class="mention" href="/mithali-zs">@mithali-zs</a>,</p>
<blockquote>
<p>when I re run the pipeline, it fails at Setup YugabyteDB cluster as it complains that the port is already being used.</p>
</blockquote>
<p>Looks you are using self-hosted runner since hosted runner will be discarded after workflow complete.<br>
In this case, <strong>at the end</strong> of the job, you can manually turnoff the container, it will release the ports as well.<br>
<img src="https://user-images.githubusercontent.com/52772761/181100348-f836896d-06f3-494f-b981-319f79eda7b2.png" alt="image" data-base62-sha1="4DAmUU8eLxjbBDgMtEfhmKFlRjw" width="500" height="333"></p>
<p>Thanks</p>

**Comment by mithali-zs** [comment]
<p>Thanks for your help <a class="mention" href="/weide-zhou">@weide-zhou</a> :slight_smile:</p>
